src/safir/dependencies/arq.py

```diff
@@ -0,0 +1,592 @@
+        self, job_id: str, queue_name: Optional[str] = None
+    ) -> JobMetadata:
+        """Get metadata about a `~arq.jobs.Job`.
+
+        Parameters
+        ----------
+        job_id : `str`
+            The job's identifier. This is the same as the `JobMetadata.id`
+            attribute, provided when initially adding a job to the queue.
+        queue_name : `str`, optional
+            Name of the queue.
+
+        Returns
+        -------
+        `JobMetadata`
+            Metadata about the queued job.
+
+        Raises
+        ------
+        JobNotFound
+            Raised if the job is not found in the queue.
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    async def get_job_result(
+        self, job_id: str, queue_name: Optional[str] = None
+    ) -> JobResult:
+        """The job result, if available.
+
+        Parameters
+        ----------
+        job_id : `str`
+            The job's identifier. This is the same as the `JobMetadata.id`
+            attribute, provided when initially adding a job to the queue.
+        queue_name : `str`, optional
+            Name of the queue.
+
+        Returns
+        -------
+        `JobResult`
+            The job's result, along with metadata about the queued job.
+
+        Raises
+        ------
+        JobNotFound
+            Raised if the job is not found in the queue.
+        JobResultUnavailable
+            Raised if the job's result is unavailable for any reason.
+        """
+        raise NotImplementedError
+
+
+class RedisArqQueue(ArqQueue):
+    """A distributed queue, based on arq and Redis."""
+
+    def __init__(
+        self,
+        pool: ArqRedis,
+        *,
+        default_queue_name: str = arq_default_queue_name,
+    ) -> None:
+        super().__init__(default_queue_name=default_queue_name)
+        self._pool = pool
+
+    @classmethod
+    async def initialize(
+        cls,
+        redis_settings: RedisSettings,
+        *,
+        default_queue_name: str = arq_default_queue_name,
+    ) -> RedisArqQueue:
+        """Initialize a RedisArqQueue from Redis settings."""
+        pool = await create_pool(
+            redis_settings, default_queue_name=default_queue_name
+        )
+        return cls(pool)
+
+    @property
+    def default_queue_name(self) -> str:
+        """Name of the default queue, if the ``_queue_name`` parameter is
+        no set in method calls.
+        """
+        return self._default_queue_name
+
+    async def enqueue(
+        self,
+        task_name: str,
+        *task_args: Any,
+        _queue_name: Optional[str] = None,
+        **task_kwargs: Any,
+    ) -> JobMetadata:
+        job = await self._pool.enqueue_job(
+            task_name,
+            *task_args,
+            _queue_name=_queue_name or self.default_queue_name,
+            **task_kwargs,
+        )
+        if job:
+            return await JobMetadata.from_job(job)
+        else:
+            raise JobNotQueued
```

Comment:
It just returns `None` on failure with no meaningful error information? That's kind of rude... not sure how you should debug what failed.

Reply:
Er, maybe this isn't obvious, but I was talking about arq? The implication of this code is that the arq function you're calling might just return `None` with no other errors.

Reply:
Whoops sorry about that. I didn't catch the context of the code snippet and totally mis-read the comment.

Turns out arq's enqueue method only returns `None` if a job with that ID already exists, so I'll include that in the new exception message. This PR doesn't implement hard-coding job IDs yet though.